ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/FileOperations.java

```diff
@@ -167,10 +169,25 @@ private void writeFilesList(FileStatus[] fileStatuses, BufferedWriter writer, St
   }
 
   private FileStatus[] listFilesInDir(Path path) throws IOException {
-    return dataFileSystem.listStatus(path, p -> {
-      String name = p.getName();
-      return !name.startsWith("_") && !name.startsWith(".");
-    });
+    boolean done = false;
+    int repeat = 0;
+    FileStatus[] files = null;
+    while (!done) {
+      try {
+        files =  dataFileSystem.listStatus(path, p -> {
+          String name = p.getName();
+          return !name.startsWith("_") && !name.startsWith(".");
+        });
+        done = true;
+      } catch (IOException e) {
+        if (++repeat == MAX_IO_ERROR_RETRY) {
```

Comment:
Need post increment or else, it is retried only 4 times.
Shall use repeat++ < MAX_IO_ERROR_RETRY.

Reply:
the behavior is same as MAX_COPY_RETRY, it will try 5 times including first time without error

Reply:
ok... but suggest to use < instead of ==.